syntax = "proto3";

package auth;

service AuthService {
    rpc CheckGroup(GroupRequest) returns (GroupResponse);
    rpc GenerateGroupPassword(GenerateRequest) returns (GenerateResponse);
    rpc HealthCheck(HealthRequest) returns (HealthResponse);
}

message GroupRequest {
    string group_name = 1;
    string password_phrase = 2;
}

message GroupResponse {
    bool exists = 1;
    bool valid_password = 2;
    string group_description = 3;
    string message = 4;
}

message GenerateRequest {
    string group_name = 1;
}

message GenerateResponse {
    bool success = 1;
    string password = 2;
    string error = 3;
}

message HealthRequest {}

message HealthResponse {
    bool healthy = 1;
    string status = 2;
}